test/denormalized_has_many_test.rb

```diff
@@ -88,4 +92,10 @@ def test_saving_associated_records_should_expire_itself_and_the_parents_cache
     IdentityCache.cache.expects(:delete).with(@record.primary_cache_index_key)
     child.save!
   end
+
+  def test_fetch_association_does_not_allow_chaining
```

Comment:
I would prefer we test for the return type to be an `Array` than not to respond_to `where`.

Chaining is one of the things we want to prevent, but in the end what we really want is a raw array of objects not tied to AR.